scripts/generateMetadata.js

```diff
@@ -46,32 +53,55 @@ async function docgen(filePath) {
   }
 }
 
-const getComponentNameFromFile = (file) => file.replace(/^.*[\\/]/, '').replace(/\.js/, '');
```

Comment:
also, out of curiosity, 

why did we add so many more lines of code? what did change in the logic that now we require  all that new logic? 


before it seems there was some recursive logic, and now it seems more inlined.

can you add a little context on this change of approach? just to follow better your decision making process here

Reply:
The main motivation of this PR is to make the script work without generating error messages, which are just noises given the current state of it. What makes them just noises? Because the files where the errors are coming from, are not supposed to have metadata, so they doesn't need to be parsed by react-docgen.

So how do we exclude the unwanted files? We can add all of them to excluded paths list but they are a lot of them. Also, they are scattered among the component files and hard to identify by file name or folder they are in.

But we can identify the files that needs to be parsed easily! We can just look at the `index.js` file! That's why I think this approach is more accurate. 

The previous logic read every single file in the given folder. Yes, we had a filter on css files and test files (but flowtest files were still slipping through, but this was an easy fix). However, there are files that are not components, that are internal components or just utilities. 

Yes, I agree, the more code can introduce more confusion or bugs (that's tried to comment as much as I can 😬). Also, I am assuming and relying on the current conventions we have in the codebase. For example, the file/folder structures and certain syntax rules. For example, subcomponents are static fields of a component: `Component.Subcomponent`. But I don't think we are going to change in the foreseeable future.